### PR TITLE
Pass PostCSS Runner Guidelines

### PR DESCRIPTION
**Pull request by ai**
Fix many issues to pass PostCSS Runner Guidelines #5 

@lydell can you review it as author of `autoprefixer-brunch`?

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -18,10 +18,35 @@ module.exports = PostCSSCompiler = (function() {
...
 
 	PostCSSCompiler.prototype.optimize = function(params, callback) {
-		var result = postcss(this.processors).process(params.data).css;
-		return callback(null, result);
+		var opts = {
+			from: params.path,
+			to:   params.path,
+			map:  {
+			    inline: false,
+	            annotation: false,
```
Comment (lydell) [inline lib/index.js]: This and the next line are indented one level too much.

Comment (ai) [inline lib/index.js]: Fixed. Seems like tabs vs space issue.

**Comment by lydell** [comment]
After a quick look and comparison with autoprefixer-brunch, this looks good to me.

**Comment by ai** [comment]
@lydell thanks :)

**Comment by iamvdo** [comment]
:+1:

**Comment by lydell** [comment]
I’ve deprecated autoprefixer-brunch in favor of this plugin now that this PR has been merged :)

**Comment by ai** [comment]
^_^

**Comment by dmitrykiselyov** [comment]
@ai :+1:  but after installing the latest version (0.4.0) I have this error:
[Error: Unsupported previous source map format: {"version":3,"sources":["app/main.css"],"names":[],"mappings":"AAAA;AACA;AACA","file":"public\app.css","sourcesContent":["body {\r\n\tcolor: red;\r\n}"]}]

`brunch new https://github.com/brunch/dead-simple` ok
`npm install --save postcss-brunch` error

**Comment by ai** [comment]
Can you fix it, I am not Brunch developer at all.

**Comment by ai** [comment]
Maybe you need `toString()` at line 39.

**Comment by dmitrykiselyov** [comment]
@ai I will try :smile:

**Comment by lydell** [comment]
I just made a quick test brunch project with latest brunch and latest
postcss-brunch. To get a previous source map I also installed the latest
less-brunch.

Postcss is supposed to accept a mozilla/source-map SourceMapGenerator as a
previous source map, but the `instanceof` check postcss uses fails because
postcss and brunch use different versions of mozilla/source-map:

```
$ grep version node_modules/brunch/node_modules/source-map/package.json
"version": "0.3.0",
$ grep version node_modules/postcss-brunch/node_modules/postcss/node_modules/source-map/package.json
"version": "0.4.2",
```

Perhaps we should open an issue at mozilla/source-map just like
mozilla/source-map#147.

However, to fix this quickly we can convert the SourceMapGenerator instance to a
regular JavaScript object before passing it to postcss, by changing `params.map`
into `params.map.toJSON()`. I’ll make a PR in a few minutes.

**Comment by dmitrykiselyov** [comment]
@lydell :+1:

**Pull request merged**